data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -244,15 +267,27 @@ Map<TopicPartition, OffsetAndMetadata> getOffsetsToCommit() {
 
     @Override
     public void run() {
-        consumer.subscribe(Arrays.asList(topicName));
+        consumer.subscribe(Arrays.asList(topicName), this);
+        Set<TopicPartition> partitions = consumer.assignment();
+        synchronized (ownedPartitionsEpoch) {
```

Comment:
This is unnecessary because, other place `ownedPartitionsEpoch` is protected by `synchronized(this)`